### PR TITLE
add ut for utils dataset

### PR DESCRIPTION
**Pull request by allenhaozi**
Signed-off-by: mahao <allenhaozi@gmail.com>

### Ⅰ. Describe what this PR does
ut for dataset

### CODE REVIEW

**Comment by allenhaozi** [comment]
@TrafalgarZZZ Please review it, thank  u

File: pkg/utils/dataset_test.go [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+
+}
+
+func TestGetAccessModesOfDataset(t *testing.T) {
+
+	testCases := map[string]struct {
+		name            string
+		getName         string
+		namespace       string
+		accessModel     []v1.PersistentVolumeAccessMode
```
Comment (TrafalgarZZZ) [inline pkg/utils/dataset_test.go:118]: `accessModes` or `accessMode` instead of `accessModel` which may be a little bit confusing

**Review by TrafalgarZZZ — COMMENTED** [review]
Also, found some other outdated codes [here](https://github.com/fluid-cloudnative/fluid/pull/849/files#diff-b7effe9a5341210f82ea0914256ef1b38598a310d3bab82459c3a569e28a294aR76). I think it would be better if we replace it with `UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Spec.Mounts)`. Can you also help fix this in the PR? Thx a lot.

Others LGTM except for the `accessModel` thing

**Comment by allenhaozi** [comment]
> Found some other outdated codes [here](https://github.com/fluid-cloudnative/fluid/blob/master/pkg/utils/dataset.go#L77). I think it would be better if we replace it with `UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Spec.Mounts)`. Can you also help fix this in the PR? Thx a lot.
> 
> Others LGTM except for the `accessModel` thing

I noticed this function and thought about it but because it's in the package of Alluxio

Other runtimes follow this rule, right?

At the same time, this DataSet uses all the runtimes, right?

**Review by cheyang — APPROVED** [review]

**Review by cheyang — APPROVED** [review]

**Pull request merged**